### ISSUE TITLE
Dense eigen decomposition of sparse matrices raises a DeprecationWarning on SciPy 1.11

### ISSUE THREAD

**Issue by hodgestar** [qutip/qutip#2093]
### Bug Description

SciPy 1.11 raises the following two deprecation warnings during qutip-4.7.X test runs:
````
FAILED qutip/tests/test_sp_eigs.py::test_DenseHermValsVecs - DeprecationWarning: Keyword argument 'eigvals' is deprecated in favour of 'subset_by_index' keyword instead and will be removed in SciPy 1.12.0.
FAILED qutip/tests/test_sp_eigs.py::test_DenseValsOnly - DeprecationWarning: Keyword argument 'eigvals' is deprecated in favour of 'subset_by_index' keyword instead and will be removed in SciPy 1.12.0.
=========== 2 failed, 3250 passed, 12 skipped in 1792.40s (0:29:52) ============
```

### Code to Reproduce the Bug

```shell
pytest -W error qutip/tests/test_sp_eigs.py
```


### Code Output

```shell
FAILED qutip/tests/test_sp_eigs.py::test_DenseHermValsVecs - DeprecationWarning: Keyword argument 'eigvals' is deprecated in favour of 'subset_by_index' keyword instead and will be removed in SciPy 1.12.0.
FAILED qutip/tests/test_sp_eigs.py::test_DenseValsOnly - DeprecationWarning: Keyword argument 'eigvals' is deprecated in favour of 'subset_by_index' keyword instead and will be removed in SciPy 1.12.0.
```


### Expected Behaviour

Tests do not fail.

### Your Environment

```shell
QuTiP Version: 4.7.1
SciPy Version: 1.11
```


### Additional Context

_No response_

**Comment by hodgestar** [comment]
Being fixed in #2081 (alongside the heterodyne stochastic solver bug).

**Comment by hodgestar** [comment]
Fixed in #2081.
